Shippable is failing due to unrelated build error, it has nothing to do with the current patch:

```
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/nxp/mcux/drivers/imx/./fsl_lpuart.h: In function ‘LPUART_GetDataRegisterAddress’:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/nxp/mcux/drivers/imx/./fsl_lpuart.h:500:30: error: ‘LPUART_Type {aka struct <anonymous>}’ has no member named ‘data’
     return (uint32_t) & (base->DATA);
                              ^~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/nxp/mcux/drivers/imx/./fsl_lpuart.h: In function ‘LPUART_WriteByte’:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/nxp/mcux/drivers/imx/./fsl_lpuart.h:602:9: error: ‘LPUART_Type {aka struct <anonymous>}’ has no member named ‘data’
     base->DATA = data;
         ^~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/nxp/mcux/drivers/imx/./fsl_lpuart.h: In function ‘LPUART_ReadByte’:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/nxp/mcux/drivers/imx/./fsl_lpuart.h:624:21: error: ‘LPUART_Type {aka struct <anonymous>}’ has no member named ‘data’
         return (base->DATA & 0x7F);
                     ^~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/nxp/mcux/drivers/imx/./fsl_lpuart.h:628:20: error: ‘LPUART_Type {aka struct <anonymous>}’ has no member named ‘data’
         return base->DATA;
                    ^~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/nxp/mcux/drivers/imx/./fsl_lpuart.h:633:1: error: control reaches end of non-void function [-Werror=return-type]
 }
 ^
cc1: all warnings being treated as errors
```

--- a/tests/kernel/threads/thread_apis/src/main.c
+++ b/tests/kernel/threads/thread_apis/src/main.c
@@ -14,6 +14,9 @@
  */
 
 #include <ztest.h>
+#include <kernel_structs.h>
+#include <kernel.h>
+
 extern void test_threads_spawn_params(void);
 extern void test_threads_spawn_priority(void);
 extern void test_threads_spawn_delay(void);
@@ -58,7 +61,7 @@ void test_systhreads_idle(void)
 	k_sleep(100);
 	/** TESTPOINT: check working thread priority should */
 	zassert_true(k_thread_priority_get(k_current_get()) <
-			K_IDLE_PRIO, NULL);
+		     K_IDLE_PRIO, NULL);
 }
 
 static void customdata_entry(void *p1, void *p2, void *p3)
@@ -112,6 +115,44 @@ void test_customdata_get_set_preempt(void)
 	k_thread_abort(tid);
 }
 
+#ifndef CONFIG_ARCH_HAS_USERSPACE
+static void umode_entry(void *thread_id, void *p2, void *p3)
+{
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	if (!_is_thread_essential() &&
+	    (k_current_get() == (k_tid_t)thread_id)) {
+		ztest_test_pass();
+	} else {
+		zassert_unreachable("User thread is essential or thread"
+				    " structure is corrupted\n");
+	}
+}
+
+/**
+ * @ingroup kernel_thread_tests
+ * @brief Test k_thread_user_mode_enter() to cover when userspace
+ * is not supported/enabled
+ * @see k_thread_user_mode_enter()
+ */
+void test_user_mode(void)
+{
+	_thread_essential_set();
+
+	zassert_true(_is_thread_essential(), "Thread isn't set"
+		     " as essential\n");
+
+	k_thread_user_mode_enter((k_thread_entry_t)umode_entry,
+				 k_current_get(), NULL, NULL);
+}
+#else
+void test_user_mode(void)
+{
+	ztest_test_skip();
+}
+#endif
+
 void test_main(void)
 {
 	k_thread_access_grant(k_current_get(), &tdata, tstack, NULL);
@@ -137,7 +178,8 @@ void test_main(void)
 			 ztest_unit_test(test_systhreads_idle),
 			 ztest_unit_test(test_customdata_get_set_coop),
 			 ztest_user_unit_test(test_customdata_get_set_preempt),
-			 ztest_unit_test(test_k_thread_foreach)
+			 ztest_unit_test(test_k_thread_foreach),
+			 ztest_unit_test(test_user_mode)
 			 );
 
 	ztest_run_test_suite(threads_lifecycle);

